great_expectations/rule_based_profiler/helpers/simple_semantic_type_filter.py

```diff
@@ -60,7 +78,7 @@ def parse_semantic_domain_type_argument(
             return [semantic_type for semantic_type in [semantic_types]]
 
         if isinstance(semantic_types, list):
-            if all(isinstance(semantic_type, str) for semantic_type in semantic_types):
+            if _is_sequence_of(semantic_types, str):
```

Comment:
Use a `TypeGuard` to help with type-narrowing

https://mypy.readthedocs.io/en/stable/type_narrowing.html#typeguards-with-parameters